core/src/main/java/org/keycloak/util/JWKSUtils.java

```diff
@@ -32,12 +33,16 @@
  */
 public class JWKSUtils {
 
+    private static final Logger logger = Logger.getLogger(JWKSUtils.class);
+
     public static Map<String, PublicKey> getKeysForUse(JSONWebKeySet keySet, JWK.Use requestedUse) {
         Map<String, PublicKey> result = new HashMap<>();
 
         for (JWK jwk : keySet.getKeys()) {
             JWKParser parser = JWKParser.create(jwk);
-            if (jwk.getPublicKeyUse().equals(requestedUse.asString()) && parser.isKeyTypeSupported(jwk.getKeyType())) {
+            if (jwk.getPublicKeyUse() == null) {
+                logger.debugf("Ignoring JWK key '%s'. Missing required field 'use'.", jwk.getKeyId());
```

Comment:
I would say, it will be better to use `logger.warnf` here? It will be good to notify administrator in more visible way that key is going to be ignored. Same applies for additional `logger.debugf` messages here.

Reply:
How about a `infof` instead? In our case, we don't have control of the external identity provider where the JWK is fetched from, so a `warnf` would always log a warning with no way for us to fix it.
Also I suggest that we flip the `equals` from `jwk.getPublicKeyUse().equals(requestedUse.asString())` to `requestedUse.asString().equals(jwk.getPublicKeyUse())` to make it null safe by design. Ok?

Reply:
@knutz3n Agree with your last comment regarding `equals` . Regarding `infof`, I am not sure as info is dedicated for informative messages, but this one is more like "invalid input" state, which doesn't fall into that. So let's stick with the debug for now.

Reply:
Ok, I left the logging as a `debugf`, but flipped the equals checks.
Any more adjustments needed before a merge?

Reply:
No, everything is fine. Thanks!